### PR TITLE
Support basic auth

### LINKED ISSUE #170: support basic auth
support basic auth for safer RPC connections

### PR DESCRIPTION
**Pull request by realcodywburns**
- Moved new web3 from web3relay & sync to ethernode.js to simplify configuration.
- switch web3.js 0.19.0 >> zsl-web3.js to support ethereum, ETC, quorum, zsl-quorum & https 
- update config.json to support:
-- bool : isSSL (uses https v. http)
-- bool : unsafeTLS (connect to https w/self signed certs)
-- string : username (basic auth username)
-- string : password (basic auth password)
- create config.js in tools to consume config.json in one location

resolves #170 
incorporates #166

### CODE REVIEW

**Comment by hackmod** [comment]
I think it contains too many changes in one PR.

File: db.js [hunk 1]
```diff
@@ -66,6 +67,35 @@ var BlockStat = new Schema(
     "uncleCount": Number
 });
 
+var Logs = new Schema(
```
Comment (hackmod) [inline db.js]: not used

File: db.js [hunk 2]
```diff
@@ -66,6 +67,35 @@ var BlockStat = new Schema(
...
+    "address" : String,
+    "data" : String,
+    "topics" : String,
+    "logIndex" : Number,
+    "blockHash" : String,
+    "blockNumber" : Number
+  }
+);
+
+var transactionReceipt = new Schema(
```
Comment (hackmod) [inline db.js]: not used

File: package.json [hunk 3]
```diff
@@ -22,7 +22,8 @@
     "serve-favicon": "~2.4.5",
     "socket.io": "^2.0.4",
     "solc": "^0.4.19",
-    "web3": "~0.19.0"
+    "web3": "~0.19.0",
+    "zsl-web3.js": "^0.21.14"
```
Comment (hackmod) [inline package.json]: commit comment is not correctly described it's changes

config.json fix and `zsl-web3.js` changes are mixed should be separated.

File: tools/config.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+}
+
+// set defaults
+// set the default geth port if it's not provided
+if (!('gethPort' in config) || (typeof config.gethPort) !== 'number') {
+    config.gethPort = 8545; // default
+}
+// set the default output directory if it's not provided
+if (!('output' in config) || (typeof config.output) !== 'string') {
+    config.output = '.'; // default this directory
```
Comment (hackmod) [inline tools/config.js]: not used anywhere

**Comment by realcodywburns** [comment]
Closing, wrong PR branch pulled in. too many mods in one PR.

**Pull request closed without merging**